paasta_tools/mesos_tools.py

```diff
@@ -627,15 +627,22 @@ def get_mesos_network_for_net(net):
     return docker_mesos_net_mapping.get(net, net)
 
 
-def get_mesos_task_count_by_slave(mesos_state, pool=None):
+def get_mesos_task_count_by_slave(mesos_state, slaves_list=None, pool=None):
     """Get counts of running tasks per mesos slave. Also include separate count of chronos tasks
 
     :param mesos_state: mesos state dict
+    :param slaves_list: a list of slave dicts to count running tasks for.
     :param pool: pool of slaves to return (None means all)
-    :returns: dict of {<slave_id>:{'count': <count>, 'slave': <slave_object>, 'chronos_count':<chronos_count>}}
+    :returns: list of slave dicts {'task_count': SlaveTaskCount}
     """
     all_mesos_tasks = get_running_tasks_from_active_frameworks('')  # empty string matches all app ids
-    if pool:
+    if slaves_list:
```

Comment:
I think this function could be made slightly simpler:

lines 639-653 can be replaced with:
```
slaves = slave['id']: {'count': 0, 'slave': slave, 'chronos_count': 0} for slave in mesos_state.get('slaves', [])
if pool:
  slaves = [slave for slave in slaves if slave['attributes'].get('pool', 'default') == pool]
elif slaves:
  slave_hosts = [slave['hostname'] for slave in slaves_list]
  slaves = [slave for slave in slaves if slave['hostname'] in slave_hosts
```